README.md

```diff
@@ -120,6 +127,65 @@ By default (when `emitCss: true`) the CSS styles will be emitted into a virtual
 
 If you set `emitCss: false` and your Svelte components contain `<style>` tags, the compiler will add JavaScript that injects those styles into the page when the component is rendered. That's not the default, because it adds weight to your JavaScript, prevents styles from being fetched in parallel with your code, and can even cause CSP violations.
 
+
+## Hot module replacement
+
+This plugin currently includes HMR support via community supported [svelte-hmr](https://github.com/rixo/svelte-hmr). This will be replaced by the official implementation once it is supported by Svelte's compiler itself.
+
+What this option does is add `esm-hmr` compatible HMR handlers (see [there](https://github.com/rixo/svelte-hmr#whats-hmr-by-the-way) for an in-depth explanation) to the compiled components code.
+
+By itself, this won't give you HMR out of the box with Rollup, because Rollup doesn't come with a dev server. See bellow for some HMR solutions with Rollup, or usage in HMR-enabled Vite.
+
+Note that the `dev` compiler option is required for HMR. **If `compilerOptions.dev` is not set or falsy, the `hot` option will be disabled too.** (Which is probably what you want for a production build.)
+
+### Options
+
+For now (while using the community implementation), you can pass any of [svelte-hmr's options](https://github.com/rixo/svelte-hmr#options) to the `hot` option of this plugin.
+
+For example:
+
+```js
+svelte({
+  compilerOptions: {
+    dev: true,
+  },
+  hot: {
+    preserveLocalState: true,
+  }
+})
+```
+
+### HMR with Rollup
+
+Rollup doesn't support HMR natively, because it doesn't even come with a dev server. If you want to enable HMR in a Rollup project, you'll need a HMR plugin like [rollup-plugin-hot](https://github.com/rixo/rollup-plugin-hot).
+
+Or you can use [Nollup](https://github.com/PepsRyuu/nollup), a Rollup compatible dev server intended for faster rebuild during development (note that you'd also need a `esm-hmr` compatibility plugin with Nollup, like [this](https://github.com/rixo/rollup-plugin-hot-nollup)).
+
+### HMR with Vite
+
+This plugin can also be used directly with Vite 2, that supports a subset of Rollup plugin API, and has native HMR.
```

Comment:
This raises the question of how do you use Svelte with Vite? :smile:  Is the only way to do that today is Svite? If so, it's probably worth mentioning Svite here

Reply:
No, just dropping a Svelte plugin (this one, or the hot fork, or this branch) in Vite 2 is enough.

I followed the steps from [this comment](https://github.com/rixo/svelte-hmr/issues/24#issuecomment-764868771) to create a Vite project, which are essentially:

~~~bash
npm init @vitejs/app
    Project name: vite2-svelte-hmr
    Select a template: vue

cd vite2-svelte-hmr
yarn
yarn add -D rixo/rollup-plugin-svelte#svelte-hmr
~~~

And replace the content of `vite.config.js` (that initially contains the Vue plugin, because of the chosen template) with the config mentioned in the README.

```js
import svelte from 'rollup-plugin-svelte'

export default {
  plugins: [
    svelte({
      compilerOptions: {
        dev: true, // should be disabled for production build
      },
      hot: true,
      emitCss: false,
    }),
  ],
}
```

That's all what's needed to make it work. Not sure how well HMR does work in Vite 2, but I don't really expect specific issues. I've asked @jsprog, who's opened the Vite 2 issue at `svelte-hmr`, for some feedback after some usage.